src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AppointmentsResourceTest.kt

```diff
@@ -0,0 +1,314 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyLong
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.ArgumentMatchers.isNull
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyNoInteractions
+import org.mockito.Mockito.`when`
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.boot.test.mock.mockito.SpyBean
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod.DELETE
+import org.springframework.http.HttpMethod.GET
+import org.springframework.http.HttpMethod.PUT
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.BodyInserters
+import uk.gov.justice.hmpps.prison.aop.ProxyUserAspect
+import uk.gov.justice.hmpps.prison.api.model.ScheduledEvent
+import uk.gov.justice.hmpps.prison.api.model.bulkappointments.AppointmentDefaults
+import uk.gov.justice.hmpps.prison.api.model.bulkappointments.AppointmentDetails
+import uk.gov.justice.hmpps.prison.api.model.bulkappointments.AppointmentsToCreate
+import uk.gov.justice.hmpps.prison.repository.BookingRepository
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+import java.util.Optional
+
+class AppointmentsResourceTest : ResourceTest() {
+  @SpyBean
+  private lateinit var proxyUserAspect: ProxyUserAspect
+
+  @MockBean
+  private lateinit var bookingRepository: BookingRepository
```

Comment:
Might be able to speed up the intergration tests by making this a SpyBean and moving both spy bean into the superclass.

Reply:
I have to do another PR for additional tests so will look at that in the next PR